sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapi/AADResourceServerOAuth2AuthorizedClientRepository.java

```diff
@@ -78,120 +51,29 @@ public AADResourceServerOAuth2AuthorizedClientRepository(OAuth2AuthorizedClientS
             LOGGER.error("Not found the ClientRegistration, registrationId={}", registrationId);
             return null;
         }
-        if (AADAuthorizationGrantType.ON_BEHALF_OF.isSameGrantType(clientRegistration.getAuthorizationGrantType())) {
-            return loadOboAuthorizedClient(clientRegistration, registrationId, authentication, request);
-        } else if (AADAuthorizationGrantType.CLIENT_CREDENTIALS
-            .isSameGrantType(clientRegistration.getAuthorizationGrantType())) {
+        if (AADAuthorizationGrantType.CLIENT_CREDENTIALS
```

Comment:
@backwind1233 do we still need this authorizedClientRepository? Would it be okay if we use the one Spring Security provides (if there's one)?

Reply:
the default is AuthenticatedPrincipalOAuth2AuthorizedClientRepository, it works if we remove AADResourceServerOAuth2AuthorizedClientRepository

Reply:
I suggest we may merge the PR first and in another task remove the AADResourceServerOAuth2AuthorizedClientRepository,
since we also have 2 unit test class beased on this.